Scripts/dea_bandindices.py

```diff
@@ -0,0 +1,255 @@
+
+                  # Ferrous Minerals Ratio, Segal 1982
+                  'FMR': lambda ds: (ds.swir1 / ds.nir),
+
+                  # Iron Oxide Ratio, Segal 1982
+                  'IOR': lambda ds: (ds.red / ds.blue)
+    }
+
+    try:
+
+        # Select a water index function based on 'water_index'
+        index_func = index_dict[index]
+
+        # Rename bands to a consistent format if either 'Collection3'
+        # or 'Sentinel2' is specified by `source`
+        if source == 'LandsatCollection3':
+
+            # Dictionary mapping full data names to simpler 'red' alias names
+            bandnames_dict = {
+                'nbart_nir': 'nir',
+                'nbart_red': 'red',
+                'nbart_green': 'green',
+                'nbart_blue': 'blue',
+                'bart_swir_1': 'swir1',
+                'nbart_swir_2': 'swir2',
+                'nbar_red': 'red',
+                'nbar_green': 'green',
+                'nbar_blue': 'blue',
+                'nbar_nir': 'nir',
+                'nbar_swir_1': 'swir1',
+                'nbar_swir_2': 'swir2'
+            }
+
+            # Rename bands in dataset to use simple names (e.g. 'red')
+            bands_to_rename = {
+                a: b for a, b in bandnames_dict.items() if a in ds.variables
+            }
+
+        elif source == 'Sentinel2':
+
+            # Dictionary mapping full data names to simpler 'red' alias names
+            bandnames_dict = {
+                'nbart_red': 'red',
+                'nbart_green': 'green',
+                'nbart_blue': 'blue',
+                'nbart_nir_1': 'nir',
+                'nbart_swir_2': 'swir1',
+                'nbart_swir_3': 'swir2',
+                'nbar_red': 'red',
+                'nbar_green': 'green',
+                'nbar_blue': 'blue',
+                'nbar_nir': 'nir',
+                'nbar_swir_2': 'swir1',
+                'nbar_swir_3': 'swir2'
+            }
+
+            # Rename bands in dataset to use simple names (e.g. 'red')
+            bands_to_rename = {
+                a: b for a, b in bandnames_dict.items() if a in ds.variables
+            }
+
+        elif source == 'LandsatCollection2':
+
+            # Pass an empty dict as no bands need renaming
+            bands_to_rename = {}
+
+        # Apply water index function to data and add to input dataset. If a
+        # custom name is supplied for the output water index variable, use it.
+        if custom_varname:
+
+            # Apply function after normalising to 0.0-1.0 by dividing by 10K
+            ds[custom_varname] = index_func(
+                ds.rename(bands_to_rename) / 10000.0)
+
+        else:
+
+            # Apply function after normalising to 0.0-1.0 by dividing by 10K
+            ds[index] = index_func(ds.rename(bands_to_rename) / 10000.0)
+
+        # Return input dataset with added water index variable
+        return ds
+
+    except AttributeError as e:
+        raise Exception(
```

Comment:
typically this type of error will be considered a `ValueError` -- user supplied data doesn't match expectations of the code. But see above comment for `AttributeError` handling.

Reply:
ok so it's a bit difficult to test ahead of time which bands are expected since that depends on the type of the index being computed. So using exception handling is easier, so I would recommend only wrapping a call to `index_func` in `try..except` block. something like:

```python
try:
    v = index_func(ds.rename(bands_to_rename)/1000.0)
except AttributeError:
   raise ValueError("...message...")

ds[custom_varname or index] = v
```